drivers/io-expander/ltc4306/ltc4306.c

```diff
@@ -0,0 +1,817 @@
+int ltc4306_read_alert_logic_state(struct ltc4306_dev *dev,
+				   uint8_t alert_pin_number, bool *is_high)
+{
+	int ret;
+	uint8_t temp;
+
+	if (alert_pin_number < LTC4306_MIN_CHANNEL_INDEX
+	    || alert_pin_number > LTC4306_MAX_CHANNEL_INDEX)
+		return -EINVAL;
+
+	ret = ltc4306_read(dev, LTC4306_CTRL_REG0, &temp, 1);
+	if (ret)
+		return ret;
+
+	*is_high = (bool)no_os_field_get(LTC4306_ALERT_LOGIC(alert_pin_number), temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Reads Failed Connection Attempt bit.
+ *
+ * @param dev - The device structure.
+ * @param is_high - Stores status of Failed Connection Attempt bit
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc4306_read_failed_conn(struct ltc4306_dev *dev, bool *is_high)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = ltc4306_read(dev, LTC4306_CTRL_REG0, &temp, 1);
+	if (ret)
+		return ret;
+
+	*is_high = (bool)no_os_field_get(LTC4306_FAILED_CONN, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Reads Latched Timeout status.
+ *
+ * @param dev - The device structure.
+ * @param timed_out - Stores status of Latched Timeout bit.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc4306_get_latched_timeout(struct ltc4306_dev *dev, bool *timed_out)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = ltc4306_read(dev, LTC4306_CTRL_REG0, &temp, 1);
+	if (ret)
+		return ret;
+
+	*timed_out = (bool)no_os_field_get(LTC4306_LATCHED_TOUT, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Reads Realtime Timeout status.
+ *
+ * @param dev - The device structure.
+ * @param timed_out - Stores status of Realtime Timeout bit.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc4306_get_realtime_timeout(struct ltc4306_dev *dev, bool *timed_out)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = ltc4306_read(dev, LTC4306_CTRL_REG0, &temp, 1);
+	if (ret)
+		return ret;
+
+	*timed_out = (bool)no_os_field_get(LTC4306_REALTIME_TOUT, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief En/Disables upstream accelerator.
+ *
+ * @param dev - The device structure.
+ * @param upstream_en - Set true to enable upstream rise time accelerator, false otherwise.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc4306_set_upstream_accel(struct ltc4306_dev *dev, bool upstream_en)
+{
+	dev->upstream = upstream_en;
+
+	if (upstream_en)
```

Comment:
wondering if you can't just pass `upstream_en` directly to `ltc4306_reg_update`. this will reduce the number of function calls to 1 in this function.

Reply:
similar structure was used from previous submissions i.e., MAX9611 and MAX31865 as recommended by one of the reviewers. I can change and test the suggested revision for this submission but I might also need to do the same with the other previous submissions for uniformity.